### ISSUE TITLE
Links in descriptions are not clickable

### ISSUE THREAD

**Issue by cam156** [projecthydra/sufia#2042]
### Descriptive summary

In  Sufia master when I add a link to my description that link should be clickable.

In ScholarSphere
![screen shot 2016-05-18 at 11 27 28 am](https://cloud.githubusercontent.com/assets/1599081/15364749/b3353d1c-1ceb-11e6-927a-e8231c1541ff.png)

In Sufia
![screen shot 2016-05-18 at 11 27 23 am](https://cloud.githubusercontent.com/assets/1599081/15364741/af8ebbac-1ceb-11e6-9a03-70b8dc16f906.png)

### Expected behavior

When I add link into the text in the description of a work, I should see a clickable link on the show page.

### Actual behavior

The link is just plain text and not clickable, so I would have to copy the text and paste it into my browser.

### Steps to reproduce the behavior

1. Click on NewWork
1. Fill in Required Metadata and add a file
1. Click on Additional Metdata
1. Add a url like 'www.google.com' to the description
1. Save the work
1. View the work and I should see a clickable link in the description not just plain text.

**Comment by mjgiarlo** [comment]
Thank you for finding and reporting this, @cam156!

**Comment by atz** [comment]
If we include unqualified links, it gets sorta dicey when you think about the large number of TLDs that now exist.  For example, if the text refers to `ruby.io`, are they referring the kernel class or a website?  It *is* legitimate namespace for a site.  Why would you assume it was a website?  If you did, would you assume `http` or `https`?  Why not FTP, SFTP, IRC, SSH... or a reference to the title to a poem?

So I am against trying to do everything and interjecting presumed knowledge where it does not apply.  If we go beyond enhancing fully qualified absolute http/https URIs, we should have concretely spelled out our expectations.

**Comment by mjgiarlo** [comment]
@atz I don't think we should try to linkify everything. I think a sufficient approach would be to run the description field through the rails_autolink helper, and let it do its thing. You can see the regexes it uses here: https://github.com/tenderlove/rails_autolink/blob/master/lib/rails_autolink/helpers.rb

This will restore functionality that is working in Sufia 6.

**Comment by mtribone** [comment]
@mjgiarlo Am I looking at the right thing?

https://github.com/projecthydra/sufia/blob/64238e6f18f315cebf5d822d231f0a87298b8203/app/helpers/sufia/sufia_helper_behavior.rb#L96

**Comment by mtribone** [comment]
Got it. I needed to call `<%=iconify_auto_link(description) %>` when I updated works and collections for multivalues.

![screen shot 2016-05-19 at 9 33 05 am](https://cloud.githubusercontent.com/assets/4163828/15395234/e75cd41a-1da4-11e6-9afb-84c0bac445f1.png)

**Comment by atz** [comment]
To be clear, that (url) helper regex requires scheme-qualified URLs... except for the special case of `^www\.`, that are assumed to be http.  

```ruby
AUTO_LINK_RE = %r{
  (?: ((?:ed2k|ftp|http|https|irc|mailto|news|gopher|nntp|telnet|webcal|xmpp|callto|feed|svn|urn|aim|rsync|tag|ssh|sftp|rtsp|afs|file):)// | www\. )
  [^\s<\u00A0"]+
}ix
```

That seems like an acceptable compromise.  Note: it means `www.google.com` linkifies, and `google.com` doesn't.  Of the 30 or so tabs I have open, 2 are `www` addresses.  (None of our own servers are `www`.)  

tl;dr: include scheme in yo' links.
